### PR TITLE
Support initialization store labels

### PR DESCRIPTION
**Pull request by Yisaer**
### What problem does this PR solve? 
Close https://github.com/pingcap/tidb-operator/issues/2692

### What is changed and how does it work?
Support initialization store labels for TiKV

Related changes

 - Need to cherry-pick to the release branch

### Does this PR introduce a user-facing change?:

```release-note
Support initialize store labels for TiKVGroup and TidbCluster
```

### CODE REVIEW

File: pkg/manager/member/tikv_group_member_manager.go [hunk 1]
```diff
@@ -645,6 +645,20 @@ func getNewTiKVSetForTiKVGroup(tg *v1alpha1.TiKVGroup, tc *v1alpha1.TidbCluster,
 		})
 	}
 	tikvContainer.Env = util.AppendEnv(env, baseTiKVSpec.Env())
+	var storeLabels map[string]string
+	if tg.Spec.InitializeStoreLabels != nil {
+		storeLabels = tg.Spec.InitializeStoreLabels
+	} else {
+		storeLabels = map[string]string{}
+	}
+	storeLabels["groupName"] = tg.Name
```
Comment (Yisaer) [inline pkg/manager/member/tikv_group_member_manager.go:654]: For each TiKVGroup instance, we will set `groupName`: <tikvgroup-name> as its store label. @rleungx

File: pkg/apis/pingcap/v1alpha1/types.go [hunk 2]
```diff
@@ -393,6 +393,10 @@ type TiKVSpec struct {
 	// Config is the Configuration of tikv-servers
 	// +optional
 	Config *TiKVConfig `json:"config,omitempty"`
+
+	// InitializeStoreLabels indicates the store label set during initialization
+	// +optional
+	InitializeStoreLabels map[string]string `json:"initializeStoreLabels,omitempty"`
```
Comment (cofyc) [inline pkg/apis/pingcap/v1alpha1/types.go:399]: I would prefer a shorter name, `storeLabels` or `initStorelabels`

File: pkg/manager/member/utils.go [hunk 3]
```diff
@@ -386,3 +387,24 @@ func getTikVConfigMapForTiKVSpec(tikvSpec *v1alpha1.TiKVSpec, tc *v1alpha1.TidbC
...
+		keys = append(keys, k)
+	}
+	sort.Strings(keys)
+	for _, key := range keys {
+		v, ok := labels[key]
+		if ok {
+			s = fmt.Sprintf("%s,%s", s, fmt.Sprintf("%s=%s", key, v))
+		}
+	}
+	return s[1:]
```
Comment (cofyc) [inline pkg/manager/member/utils.go:409]:
```suggestion
    var envs []string
    for k, v := range labels {
        envs = append(envs, fmt.Sprintf("%s=%s", k, v)) 
    }   
    sort.Strings(envs)
    return strings.Join(envs, ",")
```

we should always try to use built-in functions to simplify code

btw, please add some basic units 1) nil map 2) 1 element map 3) more than 1 elements map

File: pkg/apis/pingcap/v1alpha1/types.go (hunk 2 above)
Comment (cofyc) [reply pkg/apis/pingcap/v1alpha1/types.go:399]: maybe we should this generic way to configure environments for containers

https://github.com/pingcap/tidb-operator/blob/master/pkg/apis/pingcap/v1alpha1/types.go#L711

Comment (Yisaer) [reply pkg/apis/pingcap/v1alpha1/types.go:399]: Setting store label by ENV is the way operator pass the store labels for tikv but not obvious and convenient for users, I still suggest exposing store labels by a certain atrributes.

Comment (cofyc) [reply pkg/apis/pingcap/v1alpha1/types.go:399]: I don't think it's necessary to provide a special way for this. It's simple as configuring environments for a normal pod in Kubernetes.

Comment (cofyc) [reply pkg/apis/pingcap/v1alpha1/types.go:399]: We already have too many special configurations. Anyone knowing the standard way to configure environments for containers can configure this env and any other environments for TiKV and other components. 

If we use this special field for `STORE_LABELS`, we have more things to consider, e.g. what if users set both `STORE_LABELS` env and `storeLabels` field. Should we merge them? If not, which one has a higher priority, etc. If we have a special field for this env, should we add fields for other envs if users ask?

Comment (Yisaer) [reply pkg/apis/pingcap/v1alpha1/types.go:399]: ok, I agree with setting store labels by env.

**Pull request closed without merging**